### ISSUE TITLE
[Crons] Implement a monitor checkin API endpoint

### ISSUE THREAD

**Issue by evanpurkhiser** [getsentry/relay#2101]
The goal of this is to implement a convenience API endpoint in relay for users to create monitor check-ins **without the complexities of constructing an envelope** or having to explicitly use a SDK.

### Goals

Implement a bare-bones endpoint that does the following:

### 1. Receives the minimum necessary information to generate a check-in

- `sentry_public_key`
- [`monitor_slug`](https://github.com/getsentry/relay/blob/faa1de98f6ecfa6e62cd88248913b76747f0f911/relay-monitors/src/lib.rs#L111)
- [`status`](https://github.com/getsentry/relay/blob/faa1de98f6ecfa6e62cd88248913b76747f0f911/relay-monitors/src/lib.rs#L44-L53)

### Endpoint design details

We would like to keep the URL as short as possible, since these URLs will typically end up in users shell scripts, it's ideal for it to be short and clear.

#### URL design

For the initial pass we will **only support `GET` requests**.

We will support 2 ways to authenticate check-ins

```bash
# Public key is part of the --user argument. This is desirable as it will
# keep the public key out of logs. While the key is public, it may still be
# ideal to not continually log it.
#
# Note the leading `:`! if this is left off curl will prompt for a password.
# Can also be trailing if that is simpler to implement
curl -u :5833ef4ec765499aa7c06b32a8a8879a https://oXXX.ingest.sentry.io/api/cron/{monitor_slug}/?status={status}

# For users who are not concerned about logging their key, it may be
# provided in the URL as a parameter
curl https://oXXX.ingest.sentry.io/api/cron/{monitor_slug}:5833ef4ec765499aa7c06b32a8a8879a/?status={status}
```

**Note:** When we initially discussed this, we discussed using the monitor GUID as an identifier. Because we'll be using the public key (DSN) we will have all the information we need to route the monitor check-in using it's user configured `slug` instead. This is preferred as it makes it much easier to identify at a glance which monitor this endpoint is generating check-ins for.

**Note:** For this first pass, let's just implement both types of endpoints and play with it internally a little bit and see how it feels. We can decide to keep both, remove one, or add a different style if things don't ergonomically feel right.

#### Project ID

If the project ID **absolutely must be kept** like other endpoints it should be placed after `/api/{project_id}/cron/...`. This does lengthen the URL however, so ideall this is NOT required.

#### Extra credit

To shorten the URL even more, it could be very nice to support reading the public key as a base62 token. In our examples above the token would shorten like so:

```bash
# base16: 5833ef4ec765499aa7c06b32a8a8879a
# base62: 2gR3exZ1xSi16PZIDS3cnS

curl https://oXXX.ingest.sentry.io/api/cron/{monitor_slug}:2gR3exZ1xSi16PZIDS3cnS/?status={status}

# An example
curl https://oXXX.ingest.sentry.io/api/cron/customer-invoices:2gR3exZ1xSi16PZIDS3cnS/?status=ok
```

This is not absolutely required for the initial implementation, but if it's simple enough to implement across relay for consistency, it would be nice.

#### Query parameters

* `status`
  This parameter is required and MUST be one of the valid check-in status. Otherwise 400

* `environment`
  Not required, we will default to `production` when it is not set.

#### Response

`202 Accepted` - On success. No body

`400 Bad Request` - On invalid request (status is bad, invalid public key, if we're able to detect that?). The response should be a plain textt

### 2. Constructs the appropriate monitor envelope and ingests it for processing within relay.

<details>
<summary>See the `relay-monitors` crate</summary>

https://github.com/getsentry/relay/blob/faa1de98f6ecfa6e62cd88248913b76747f0f911/relay-monitors/src/lib.rs#L105-L127

</details>

### Check-in ID

The client SDK typically handles opening / closing monitor check-ins by generating a check-in UUID. We do not want to do that here. Instead we will support a similar approach to the existing legacy endpoints, where a sentinel value `latest` can be provided as the `check_in_id`.

This is not yet supported in the monitor consumer (https://github.com/getsentry/sentry/issues/49664). This **will be required to be completed** before we can fully end to end test this endpoint in relay.

## Future work

As mentioned earlier, we should keep in mind that in the near future we would like to expand this endpoint. Since we would like to deprecate the old endpoints in sentry, we will need to support all the features they do.

Likely we will support `POST` requests with a JSON body to allow for:

 * Monitor upserts
 * Check-in ID specification
 * Generating checkins and responding on the API

Since we will just be initially implementing a GET, this should leave things pretty open ended.

**Comment by mitsuhiko** [comment]
I implemented the basics in https://github.com/getsentry/relay/pull/2153. Maybe you can take it from there. Note that if no `check_in_id` is supplied it will now send the nil UUID to the consumer. So the consumer needs to be modified before this merges.
